### DISCUSSION TITLE
Add a Slide at a specific index or Move a Slide

### DISCUSSION THREAD

**Discussion by Luigicube** [ShapeCrawler/ShapeCrawler#642] [Q&A]
Hi,
Is there an option to either _add a Slide at a specific index_ or _move a Slide_? I'm currently stuck with adding slides at the end of my document but that's not what I want.

**Comment by Luigicube** [comment]
_Slides.Insert_ it is. Sorry for the thread, FYI I did not see that in any documentation.

**Reply by Luigicube** [reply]
Though I'm not sure if the workflow I use currently is correct:

1. Adding an empty slide (last position)
2. Inserting the slide at my desired position (= Copying)
3. Removing the slide at the last position

**Reply by AdamShakhabov** [reply]
Hello.

You can use the setter of property `ISlide.Number`:

```c#
var pres = new Presentation(@"c:\slide-moving.pptx");
var numSlides = pres.Slides.Count;

pres.Slides[0].Number = numSlides;

pres.Save();
```

**Reply by Luigicube** [reply]
Thanks, it works.
